code/game/objects/items/toys/mech_toys.dm

```diff
@@ -0,0 +1,642 @@
+  */
+/obj/item/toy/mecha/attack(mob/living/carbon/human/target, mob/living/carbon/human/user)
+	if(target == user)
+		to_chat(user, "<span class='notice'>Target another toy mech if you want to start a battle with yourself.</span>")
+		return
+	else if(user.a_intent != I_HURT)
+		if(wants_to_battle) //prevent spamming someone with offers
+			to_chat(user, "<span class='notice'>You already are offering battle to someone!</span>")
+			return
+		if(!check_battle_start(user)) //if the user's mech isn't ready, don't bother checking
+			return
+
+		for(var/obj/item/I in target.get_all_held_items())
+			if(istype(I, /obj/item/toy/mecha)) //if you attack someone with a mech who's also holding a mech, offer to battle them
+				var/obj/item/toy/mecha/M = I
+				if(!M.check_battle_start(target, null, user)) //check if the attacker mech is ready
+					break
+
+				//slap them with the metaphorical white glove
+				if(M.wants_to_battle) //if the target mech wants to battle, initiate the battle from their POV
+					mecha_brawl(M, target, user) //P = defender's mech / SRC = attacker's mech / target = defender / user = attacker
+					M.wants_to_battle = FALSE
+					return
+
+		//extend the offer of battle to the other mech
+		var/datum/gender/T = gender_datums[user.get_visible_gender()]
+		to_chat(user, "<span class='notice'>You offer battle to [target.name]!</span>")
+		to_chat(target, "<span class='notice'><b>[user.name] wants to battle with [T.His] [name]!</b> <i>Attack them with a toy mech to initiate combat.</i></span>")
+		wants_to_battle = TRUE
+		addtimer(CALLBACK(src, .proc/withdraw_offer, user), 6 SECONDS)
+		return
+
+	..()
+
+/**
+  * Overrides attack_tk - Sorry, you have to be face to face to initiate a battle, it's good sportsmanship
+  */
+/obj/item/toy/mecha/attack_tk(mob/user)
+	if(timer < world.time)
+		to_chat(user, "<span class='notice'>You telekinetically play with [src].</span>")
+		timer = world.time + cooldown
+		playsound(user, 'sound/mecha/mechstep.ogg', 20, TRUE)
+
+/**
+  * Resets the request for battle.
+  *
+  * For use in a timer, this proc resets the wants_to_battle variable after a short period.
+  * Arguments:
+  * * user - the user wanting to do battle
+  */
+/obj/item/toy/mecha/proc/withdraw_offer(mob/living/carbon/user)
+	if(wants_to_battle)
+		wants_to_battle = FALSE
+		to_chat(user, "<span class='notice'>You get the feeling they don't want to battle.</span>")
+/**
+  * Starts a battle, toy mech vs player. Player... doesn't win. Commented out for now as suicide_act is not physically doable.
+  */
+/obj/item/toy/mecha/suicide_act(mob/living/carbon/user)
+	if(in_combat)
+		to_chat(user, "<span class='notice'>[src] is in battle, let it finish first.</span>")
+		return
+	
+	var/datum/gender/T = gender_datums[user.get_visible_gender()]
+	user.visible_message("<span class='suicide'>[user] begins a fight [T.His] can't win with [src]! It looks like [T.His] trying to commit suicide!</span>")
```

Comment:
I know that this is going to go unused but the lack of punctuation after "begins a fight" is gonna bother me.